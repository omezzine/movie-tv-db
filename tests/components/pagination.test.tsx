import { render, screen, fireEvent } from '@testing-library/react';
import { vi } from 'vitest';
import React from 'react';
import Pagination from '../../src/components/pagination';
import renderer from 'react-test-renderer';

describe('Pagination Component', () => {
  test('Should match snapshot', () => {
    const pagination = renderer
      .create(<Pagination pageCount={10} currentPage={'1'} offset={1} onChange={() => vi.fn()} />)
      .toJSON();
    expect(pagination).toMatchSnapshot();
  });

  it('display Pagination', async () => {
    const onPaginationCHange = vi.fn();
    const { container } = render(
      <Pagination pageCount={10} currentPage={'1'} offset={1} onChange={onPaginationCHange} />,
    );
    // screen.debug();
    fireEvent.click(container.querySelector('a[rel="next"]') as HTMLElement, {});
    expect(onPaginationCHange).toHaveBeenCalledWith('2');
    fireEvent.click(container.querySelector('a[rel="next"]') as HTMLElement, {});
    expect(onPaginationCHange).toHaveBeenCalledWith('3');
    fireEvent.click(container.querySelector('a[rel="prev"]') as HTMLElement, {});
    expect(onPaginationCHange).toHaveBeenCalledWith('2');
  });

  test('Should have selected page', () => {
    const onPaginationCHange = vi.fn();
    const { container } = render(
      <Pagination pageCount={10} currentPage={'5'} offset={1} onChange={onPaginationCHange} />,
    );
    fireEvent.click(container.querySelector('a[rel="next"]') as HTMLElement, {});
    expect(onPaginationCHange).toHaveBeenCalledWith('6');
  });
});
