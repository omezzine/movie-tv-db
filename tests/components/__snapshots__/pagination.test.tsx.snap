// Vitest Snapshot v1

exports[`Pagination Component > Should match snapshot 1`] = `
<ul
  aria-label="Pagination"
  className="pagination"
  role="navigation"
>
  <li
    className="page-item disabled"
  >
    <a
      aria-disabled="true"
      aria-label="Previous page"
      className="page-link "
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="prev"
      role="button"
      tabIndex="-1"
    >
      &lt; previous
    </a>
  </li>
  <li
    className="page-item active"
  >
    <a
      aria-current="page"
      aria-label="Page 1 is your current page"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="canonical"
      role="button"
      tabIndex="-1"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-current={null}
      aria-label="Page 2"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="next"
      role="button"
      tabIndex="0"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-current={null}
      aria-label="Page 3"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-current={null}
      aria-label="Page 9"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      9
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-current={null}
      aria-label="Page 10"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-disabled="false"
      aria-label="Next page"
      className="page-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="next"
      role="button"
      tabIndex="0"
    >
      next &gt;
    </a>
  </li>
</ul>
`;
